src/core/shape_inference/include/scatter_elements_update_shape_inference.hpp

```diff
@@ -9,13 +9,13 @@
 #include "utils.hpp"
 
 namespace ov {
-namespace op {
-namespace v3 {
 
+namespace op {
+namespace {
```

Comment:
Cosmetic make add name for namespace like `scatter_el_update` and then `shape_infer_impl`->`shape_infer`.
Or maybe introduce the Scatter base class and convert this common shape infer to 

```
namespace ov{
namespace op{
shape_infer(const util::ScatterElementUpdateBase* op...
```

Then the template wrappers for v3,v12 I think could be removed and Scatters can call common function as they will be children classes.

Maybe it will help with implementation of evaluates functions also (for reduction mode NONE I think it works as v3).